modules/peers.js

```diff
@@ -79,38 +221,14 @@ __private.updatePeersList = function (cb) {
 	}
 
 	function pickPeers (peers, waterCb) {
-		// Protect removed nodes from overflow
-		if (removed.length > 100) {
-			removed = [];
-		}
-
-		library.logger.debug('Removed peers: ' + removed.length);
-
-		// Pick peers
-		//
-		// * Removing unacceptable peers
-		// * Removing nodes not behaving well
-		var picked = self.acceptable(peers).filter(function (peer) {
-			return removed.indexOf(peer.ip);
-		});
+		var picked = self.acceptable(peers);
 
-		// Drop one random peer from removed array to give them a chance.
-		// This mitigates the issue that a node could be removed forever if it was offline for long.
-		// This is not harmful for the node, but prevents network from shrinking, increasing noise.
-		// To fine tune: decreasing random value threshold -> reduce noise.
-		if (Math.random() < 0.5) { // Every 60/0.5 = 120s
-			// Remove the first element,
-			// i.e. the one that have been placed first.
-			removed.shift();
-			removed.pop();
-		}
-
-		library.logger.debug(['Picked', picked.length, 'of', peers.length, 'peers'].join(' '));
+		library.logger.debug('Picked ' + picked.length + ' of ' + peers.length + ' peers');
```

Comment:
Can we stick to joining array for now? This standard set throughout. We can change all separately if deemed better.